### PR TITLE
docs: add options snippets to api docs

### PR DESCRIPTION
**Pull request by debs-obrien**

### CODE REVIEW

File: docs/src/test-api/class-testoptions.md [hunk 1]
```diff
@@ -35,15 +35,40 @@ test('my portrait test', async ({ page }) => {
...
 
+**Usage**
+
+```js
+import { defineConfig, devices } from '@playwright/test';
+
+export default defineConfig({
+  use: {
+    /* Base URL to use in actions like `await page.goto('/')`. */
+    baseURL: process.env.PLAYWRIGHT_TEST_BASE_URL || 'http://localhost:3000',
```
Comment (dgozman) [inline docs/src/test-api/class-testoptions.md:59]:
```suggestion
    baseURL: 'http://localhost:3000',
```

File: docs/src/test-api/class-testoptions.md [hunk 2]
```diff
@@ -63,24 +88,88 @@ Default timeout for each Playwright action in milliseconds, defaults to 0 (no ti
...
 * since: v1.10
 
+**Usage**
+
+```js
+import { defineConfig } from '@playwright/test';
+
+export default defineConfig({
+  use: {
+    channel: 'chrome',
```
Comment (dgozman) [inline docs/src/test-api/class-testoptions.md:131]: This one should be in a project for sure.

File: docs/src/test-api/class-testoptions.md [hunk 3]
```diff
@@ -63,24 +88,88 @@ Default timeout for each Playwright action in milliseconds, defaults to 0 (no ti
...
+
+**Usage**
+
+```js
+import { defineConfig } from '@playwright/test';
+
+export default defineConfig({
+  use: {
+    connectOptions: {
+      timeout: 3000
```
Comment (dgozman) [inline docs/src/test-api/class-testoptions.md:167]:
```suggestion
      wsEndpoint: 'ws://localhost:5678',
```

File: docs/src/test-api/class-testoptions.md [hunk 4]
```diff
@@ -89,42 +178,195 @@ When connect options are specified, default [`property: Fixtures.browser`], [`pr
...
 
+**Usage**
+
+```js
+import { defineConfig } from '@playwright/test';
+
+export default defineConfig({
+  use: {
+    contextOptions: {
+      viewport: viewport: { width: 1280, height: 720 },
```
Comment (dgozman) [inline docs/src/test-api/class-testoptions.md:189]:
```suggestion
      reducedMotion: 'reduce',
```

File: docs/src/test-api/class-testoptions.md [hunk 5]
```diff
@@ -89,42 +178,195 @@ When connect options are specified, default [`property: Fixtures.browser`], [`pr
...
 * since: v1.10
 
+**Usage**
+
+```js
+import { defineConfig } from '@playwright/test';
+
+export default defineConfig({
+  use: {
+    deviceScaleFactor: 2
```
Comment (dgozman) [inline docs/src/test-api/class-testoptions.md:205]: This should also be in a project. We can also say "use device" instead and link to devices doc.

File: docs/src/test-api/class-testoptions.md [hunk 6]
```diff
@@ -89,42 +178,195 @@ When connect options are specified, default [`property: Fixtures.browser`], [`pr
...
 * since: v1.10
 
+**Usage**
+
+```js
+import { defineConfig } from '@playwright/test';
+
+export default defineConfig({
+  use: {
+    hasTouch: true
```
Comment (dgozman) [inline docs/src/test-api/class-testoptions.md:252]: Point to devices.

File: docs/src/test-api/class-testoptions.md [hunk 7]
```diff
@@ -89,42 +178,195 @@ When connect options are specified, default [`property: Fixtures.browser`], [`pr
...
 * since: v1.10
 
+**Usage**
+
+```js
+import { defineConfig } from '@playwright/test';
+
+export default defineConfig({
+  use: {
+    isMobile: false,
```
Comment (dgozman) [inline docs/src/test-api/class-testoptions.md:315]: Point to devices.

File: docs/src/test-api/class-testoptions.md [hunk 8]
```diff
@@ -89,42 +178,195 @@ When connect options are specified, default [`property: Fixtures.browser`], [`pr
...
 
+**Usage**
+
+```js
+import { defineConfig } from '@playwright/test';
+
+export default defineConfig({
+  use: {
+    launchOptions: {
+      slowMo: 50,
```
Comment (dgozman) [inline docs/src/test-api/class-testoptions.md:349]:
```suggestion
      args: ['--load-extension=/path/to/extension'],
```

File: docs/src/test-api/class-testoptions.md [hunk 9]
```diff
@@ -89,42 +178,195 @@ When connect options are specified, default [`property: Fixtures.browser`], [`pr
...
 Options used to launch the browser, as passed to [`method: BrowserType.launch`]. Specific options [`property: TestOptions.headless`] and [`property: TestOptions.channel`] take priority over this.
 
+**Usage**
+
+```js
+import { defineConfig } from '@playwright/test';
+
+export default defineConfig({
+  use: {
+    launchOptions: {
```
Comment (dgozman) [inline docs/src/test-api/class-testoptions.md:348]: Put this to a "chromium" project.

File: docs/src/test-api/class-testoptions.md [hunk 10]
```diff
@@ -185,11 +526,35 @@ Whether to record trace for each test. Defaults to `'off'`.
...
 * since: v1.10
 
+**Usage**
+
+```js
+import { defineConfig } from '@playwright/test';
+
+export default defineConfig({
+  use: {
+    userAgent: 'some custom ua',
```
Comment (dgozman) [inline docs/src/test-api/class-testoptions.md:553]: Point to devices.

File: docs/src/test-api/class-testoptions.md (hunk 9 above)
Comment (mxschmitt) [reply docs/src/test-api/class-testoptions.md:350]: would --load-extension work on a normal launch()? I thought this works only with launchPersistentContext.

Comment (dgozman) [reply docs/src/test-api/class-testoptions.md]: Good point. We can use `--start-maximized` as an example.

**Review by dgozman — APPROVED** [review]

**Pull request merged**